### ISSUE TITLE
Compatibility with kubernetes 1.22

### ISSUE THREAD

**Issue by fredleger** [compumike/hairpin-proxy#20]
Seems that does not work anymore since k8s 1.22:

```
I, [2023-04-28T07:30:06.221620 #1]  INFO -- : Starting in CoreDNS mode. (Indended to be run as a Deployment: one instance per cluster.)
I, [2023-04-28T07:30:06.316336 #1]  INFO -- : Starting main_loop with 15s polling interval.
I, [2023-04-28T07:30:06.316833 #1]  INFO -- : Polling all Ingress resources and CoreDNS configuration...
W, [2023-04-28T07:30:06.518745 #1]  WARN -- : Warning: Unable to list ingresses in extensions/v1beta1
```

Which seems logical. Is there any chance to see compatibility ?

**Comment by Vaskata84** [comment]
Seems that does not work anymore since k8s 1.25:   Any solution to the problem???
│
 I, [2023-05-05T08:04:35.184575 #1]  INFO -- : Polling all Ingress resources and CoreDNS configuration...                                                                                                │
│ W, [2023-05-05T08:04:35.186571 #1]  WARN -- : Warning: Unable to list ingresses in extensions/v1beta1

**Comment by fredleger** [comment]
@Vaskata84 it's really since 1.22 that ingresses are not under `extensions/v1beta1`: https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#api-changes

But that is a detail the thing is that this project seems unmaintaned

**Comment by Vaskata84** [comment]
> 
how do you manage without it, how does cert-manager work for you

**Comment by Vaskata84** [comment]
│ E0505 10:43:25.963111       1 sync.go:190] cert-manager/challenges "msg"="propagation check failed" "error"="wrong status code '400', expected '200'" "dnsName"="zzz.example.com" "resource_kind"="Chal │

**Comment by fredleger** [comment]
i started a fork here : https://github.com/webofmars/hairpin-proxy
try to see if it can help you but without guarantee

**Comment by Vaskata84** [comment]
> i started a fork here : https://github.com/webofmars/hairpin-proxy try to see if it can help you but without guarantee

I don't understand how to install it

**Comment by fredleger** [comment]
`kubectl apply -f deploy.yaml`

**Comment by Vaskata84** [comment]
pod reboots constantly

**Comment by fredleger** [comment]
> pod reboots constantly

can you provide more details ? kubectl describe pod <pod-name> output ? kubectl logs <pod name>

**Comment by Vaskata84** [comment]
Does it work for you?

**Comment by renanramonh** [comment]
I was having same problem with k3s and tested the repo from @fredleger. It worked very well for me.

**Comment by DaanSelen** [comment]
> i started a fork here : https://github.com/webofmars/hairpin-proxy try to see if it can help you but without guarantee

Is this still maintained?

**Comment by antonengelhardt** [comment]
No, last commit 4 years ago :(

**Comment by DaanSelen** [comment]
> No, last commit 4 years ago :(

I've started to maintain it however possible on my own. Doing this in a private repo now. Is there still interest in this project?
I made it also check IngressRoutes.

**Comment by antonengelhardt** [comment]
Actually i am still having the problem that hairpin proxy initially promised to solve. Im not sure if it is related to the problem the author of this issue mentioned. Im getting the error `failed to perform self check GET request ..... EOF`. Using digitial ocean cluster with hairpin proxy and load balance in digital ocean has proxy protocol enabled.

**Comment by DaanSelen** [comment]
> Actually i am still having the problem that hairpin proxy initially promised to solve. Im not sure if it is related to the problem the author of this issue mentioned. Im getting the error `failed to perform self check GET request ..... EOF`. Using digitial ocean cluster with hairpin proxy and load balance in digital ocean has proxy protocol enabled.

Does it rewrite the CoreDNS routes correctly?

**Comment by antonengelhardt** [comment]
How can i check?

**Comment by DaanSelen** [comment]
> How can i check?

`kubectl get configmap -n kube-system coredns` I believe.
You can check if its called `coredns` with `kubectl get configmap -n kube-system` (If configured 'defaultly'`.

**Comment by antonengelhardt** [comment]
Yes it has rewritten it. Do you know my any chance how often it syncs it?

**Comment by DaanSelen** [comment]
> Yes it has rewritten it. Do you know my any chance how often it syncs it?

You should see it sync in the logs of the controller pod.
Let me check the code.

https://github.com/compumike/hairpin-proxy/blob/10ef5fd0ed8b424dc1f172ad35d7f0e6d6b704b3/hairpin-proxy-controller/src/main.rb#L12C56-L12C67 Every 15 sec.

**Comment by antonengelhardt** [comment]
Okay got it. Its very strange. 

I have tried to find the solution to my problem. Sometimes i need to add `acme.cert-manager.io/http01-edit-in-place: "true"` or `cert-manager.io/issue-temporary-certificate: "true"`. Sometimes i need to add the ingress that i want to use later on. 

The problem is that i did not find the solution because everytime i test it, the solution is different xD

**Comment by DaanSelen** [comment]
> Okay got it. Its very strange.
> 
> I have tried to find the solution to my problem. Sometimes i need to add `acme.cert-manager.io/http01-edit-in-place: "true"` or `cert-manager.io/issue-temporary-certificate: "true"`. Sometimes i need to add the ingress that i want to use later on.
> 
> The problem is that i did not find the solution because everytime i test it, the solution is different xD

Seems like something is wrong in your configuration.

**Comment by antonengelhardt** [comment]
Do you have any idea what that could be? It still works but only after some guessing and trial & error.

**Comment by DaanSelen** [comment]
With these details I have no idea, what are the logs, what are you doing? Perhaps mail me then dselen@nerthus.nl

**Comment by fredleger** [comment]
> No, last commit 4 years ago :(

I did not have the need/time to maintain it.
But i'm really ok to take PRs on it instead to all have our private forks.
I can even transfer the repo if you are up to become the "maintainer" of it.

just tell me

**Comment by fredleger** [comment]
At the time of my fork the issue was the original maintainer did not answer to PRs or issues. If that changed, I can consider merging to the original repo as well

**Comment by DaanSelen** [comment]
What even is the original repo?
